tests/test_dcgm_monitor.py

```diff
@@ -61,7 +61,7 @@ def test_record_memory(self):
 
         # Assert instance types
         for record in records:
-            self.assertIsInstance(record.device(), GPUDevice)
```

Comment:
Why the device is no longer a GPUDevice?

Reply:
The records now just store the UUID. This makes it so that only the GPUDeviceFactory creates and keeps track of GPUDevices, and to prevent needing to get extra information in the remote monitor/dcgm monitor.